src/wallet/wallet_state_manager.py

```diff
@@ -256,8 +276,36 @@ async def create_more_puzzle_hashes(self, from_zero: bool = False):
                 start_index = 0
 
             for index in range(start_index, unused + to_generate):
+                if target_wallet.wallet_info.type == WalletType.RATE_LIMITED:
```

Comment:
I'll add a property to wallets, hd_support: bool. 
If it's True, WSM does this kind of derivation for the wallet. 
If False, wallet should create a puzzle_hash and store it to puzzle_store itself.